src/System.Memory/tests/Span/CopyTo.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.SpanTests
+{
+    public static partial class SpanTests
+    {
+        [Fact]
+        public static void TryCopyTo()
+        {
+            int[] src = { 1, 2, 3 };
+            int[] dst = { 99, 100, 101 };
+
+            Span<int> srcSpan = new Span<int>(src);
+            bool success = srcSpan.TryCopyTo(dst);
+            Assert.True(success);
+            Assert.Equal<int>(src, dst);
+        }
+
+        [Fact]
+        public static void TryCopyToLonger()
+        {
+            int[] src = { 1, 2, 3 };
+            int[] dst = { 99, 100, 101, 102 };
+
+            Span<int> srcSpan = new Span<int>(src);
+            bool success = srcSpan.TryCopyTo(dst);
+            Assert.True(success);
+            int[] expected = { 1, 2, 3, 102 };
+            Assert.Equal<int>(expected, dst);
+        }
+
+        [Fact]
+        public static void TryCopyToShorter()
+        {
+            int[] src = { 1, 2, 3 };
+            int[] dst = { 99, 100 };
+
+            Span<int> srcSpan = new Span<int>(src);
+            bool success = srcSpan.TryCopyTo(dst);
+            Assert.False(success);
+            int[] expected = { 99, 100 };
+            Assert.Equal<int>(expected, dst);  // TryCopyTo() checks for sufficient space before doing any copying.
+        }
+
+        [Fact]
+        public static void CopyToShorter()
+        {
+            int[] src = { 1, 2, 3 };
+            int[] dst = { 99, 100 };
+
+            Span<int> srcSpan = new Span<int>(src);
+            Assert.Throws<ArgumentException>(() => srcSpan.CopyTo(dst));
+            int[] expected = { 99, 100 };
+            Assert.Equal<int>(expected, dst);  // CopyTo() checks for sufficient space before doing any copying.
+        }
+
+        [Fact]
+        public static void Overlapping1()
+        {
+            int[] a = { 90, 91, 92, 93, 94, 95, 96, 97 };
+
+            Span<int> src = new Span<int>(a, 1, 6);
+            Span<int> dst = new Span<int>(a, 2, 6);
+            src.CopyTo(dst);
+
+            int[] expected = { 90, 91, 91, 92, 93, 94, 95, 96 };
+            Assert.Equal<int>(expected, a);
+        }
+
+        [Fact]
+        public static void Overlapping2()
```

Comment:
Also, i don't think not using theories scales past span. Many other APIs that we will be adding (e.g. formatting) are so much more complicated than spans that we won't be able to avoid using theories.